Managers/UTMVirtualMachine+Drives.m

```diff
@@ -87,12 +87,16 @@ - (BOOL)changeMediumForDrive:(UTMDrive *)drive url:(NSURL *)url error:(NSError *
         }
         return NO;
     }
-    NSString *oldPath = [self.viewState pathForRemovableDrive:drive.name];
-    [self.viewState setBookmark:bookmark path:url.path forRemovableDrive:drive.name persistent:NO];
     if (!self.qemu.isConnected) {
+        [self.viewState setBookmark:bookmark path:url.path forRemovableDrive:drive.name persistent:YES];
```

Comment:
This still won't work for macOS. To get a persistent bookmark, you need to get it from the helper process, which at this point isn't launched yet.

We can make it so it saves the settings only for iOS (which doesn't have this requirement) but it would be inconsistent with macOS. Or we can refactor the code so it will always call `[self.system accessDataWithBookmark]`.

Reply:
Would it work if I change it to save just the drive path on macOS:

```objc
    if (!self.qemu.isConnected) {
#if TARGET_OS_IPHONE
        [self.viewState setBookmark:bookmark path:url.path forRemovableDrive:drive.name persistent:YES];
        [self saveViewState];
#else
        [self.configuration setImagePath:url.path forIndex:drive.index];
#endif
        return YES; // not ready yet
    } else {
```

After QEMU connects, in `restoreRemovableDrivesFromBookmarks` I could create a bookmark from the saved path.

Reply:
For consistency purposes I would prefer the behaviour be consistent across macOS and iOS even if it's not ideal.

Reply:
With the fix I suggested it would achieve consistent results on both platforms, even though the behind the scenes is slightly divergent. Is that not acceptable?

Reply:
Sure I’m fine with it but can you make sure to test everything on both iOS and macOS?

1. Load image and exit
2. Load image and start
3. Start and change image and exit
4. Start and don’t change image and exit
5. Start and clear and exit
6. Second launch, load and exit
7. Second launch, use existing and start
8. Second launch, clear and exit
9. Second launch, clear and start

And make sure it works as expected.

Reply:
Unfortunately, I can't get UTM to work on my MacBook Air (Big Sur 11.1 Beta). I can't start any VMs because it can't connect to the QEMU instance (console logs many _SPICE port not in use_ messages). So I haven't made any progress on this yet.

Reply:
Sorry, I don't understand your test notes. (start/launch?)

> 1. Load image and exit
> 2. Load image and start
> 3. Start and change image and exit
> 4. Start and don’t change image and exit
> 5. Start and clear and exit
> 6. Second launch, load and exit
> 7. Second launch, use existing and start
> 8. Second launch, clear and exit
> 9. Second launch, clear and start

Here's all the test scenarios that I tested on my machine. I begin with a x86 VM created that has only a removable CD drive with no image inserted (no other boot device). Each following test continues with the `.utm` state after the previous test. Observations are in (brackets).

- start UTM, Browse for an image, exit UTM, start UTM, Start VM (CD loads correctly), exit UTM
- start UTM (image inserted), clear image, exit UTM, start UTM (no image), exit UTM
- start UTM, browse for a different image, exit UTM, start UTM, browse for first image, Start VM (CD loads correctly), exit UTM
- start UTM, start VM (CD loads correctly), eject CD, exit UTM, start UTM (no image), exit UTM
- start UTM, start VM (no bootable medium), browse for image, reset VM (CD loads correctly), exit UTM

All of the above worked as expected (macOS 11.1). The new code that makes the image persistence work is in [UTMVirtualMachine+Drives.m](https://github.com/conath/UTM/blob/8a5f2060a267fe80b0fdcd9789c747f503ee249f/Managers/UTMVirtualMachine%2BDrives.m#L145).